### PR TITLE
Add Spark and Ray job CRD protos from v2beta1

### PR DESCRIPTION
**Pull request by mincai**
Add Spark and Ray job CRD protos from v2beta1 to v1 unchanged. 

A few TODO items:
- Replace PodSpec with k8s/api/corev1/ PodTemplate
- Split RayJob to RayJob and RayCluster

### CODE REVIEW

File: proto/api/v2/pod.proto [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  // Custom docker image name
+  string image = 3;
+
+  // Environment variables
+  repeated Environment env = 4;
+
+  // Security token used to read/write to secure HDFS.
+  // TODO: replace this with K8s or Peloton Secrets
+  string hdfs_delegation_token = 5;
```
Comment (badcount) [inline proto/api/v2/pod.proto:60]: This is uber specific, should we remove this?

Comment (mincai) [reply proto/api/v2/pod.proto:60]: Yes. but prefer to do that in next few diffs.

Comment (mincai) [reply proto/api/v2/pod.proto:60]: Actually we will replace the whole PodSpec with PodTemplate in k8s api. 

https://github.com/kubernetes/api/blob/master/core/v1/generated.proto#L4651

**Review by badcount — APPROVED** [review]

File: proto/api/v2/spark_job.proto [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  repeated string files = 2;
+
+  // PyFiles is a list of Python files the Spark application depends on.
+  repeated string py_files = 3;
+}
+
+// Defines the status of a Spark job
+message SparkJobStatus {
+  // Deprecated - use status_conditions instead
+  // Conditions that describe states
```
Comment (bhjoshi1) [inline proto/api/v2/spark_job.proto:87]: Should we remove all the fields that are marked as Deprecated?

**Pull request merged**